packages/aragon-wrapper/src/index.js

```diff
@@ -376,8 +377,51 @@ export default class Aragon {
       ))
     )
 
+    // SetApp events are emitted when apps are installed and upgraded
+    // These may modify the implementation addresses of the proxies (modifying their behaviour), so
+    // we invalidate any caching we've done
+    const updatedApps$ = this.kernelProxy
+      // Override events subscription with empty options to subscribe from latest block
+      .events('SetApp', {})
+      .pipe(
+        // Merge with latest value of installedApps$ so we can return the full list of apps
+        withLatestFrom(
+          installedApps$,
+          function updateApps (setAppEvent, apps) {
+            const { appId: setAppId } = setAppEvent.returnValues
+            return apps.map(async (app) => {
+              if (app.appId !== setAppId) {
+                return app
+              }
+
+              let proxyValues
+              try {
+                proxyValues = await proxyContractValueCache.request(
+                  app.proxyAddress,
+                  true // force cache invalidation
+                )
+              } catch (_) {}
+
+              return {
+                ...app,
+                ...proxyValues,
+                updated: true
+              }
+            })
+          }
+        ),
+        // Switch to resolved array of promises
+        switchMap(Promise.all)
```

Comment:
The `switchMap` here will call Promise.all on every new array of promises returned from the previous operator (`withLatestFrom`) and emit the resolved values of `Promise.all`. Why however do we use `switchMap` over `mergeMap`?

Can't we have a situation in which two `SetApp` events are emitted and we want the changes of both?

Reply:
> Can't we have a situation in which two SetApp events are emitted and we want the changes of both?

Hmm, yes, this is true. I had thought that `mergeMap()` and `switchMap()` are synonymous when it comes to single-emission observables being returned from their operators, but that's not true when the first event is still waiting to get processed when the second event fires.

There are still cases where you'd only be interested in the latest event, e.g. how the `appIdentifiers` are set up, but I think all our uses of `switchMap()` should be switched to `mergeMap()` 👍.

Reply:
See eaabdec, the decision to use `switchMap()` or `mergeMap()`, as I understand it, is if we care about the throttled values.

Reply:
So I figured this one out; the one we usually want with promises is actually [`concatMap()`](https://rxjs-dev.firebaseapp.com/api/operators/concatMap) because we'd like to queue each promise's resolution behind each other (rather than firing at will or throttling).